src/GraphQL.Conventions/Adapters/GraphTypeAdapter.cs

```diff
@@ -29,25 +29,45 @@ public ISchema DeriveSchema(GraphSchemaInfo schemaInfo)
             var interfaces = types
                 .Where(t => t.IsInterfaceType && !t.IsInputType && t.PossibleTypes.Any())
                 .GroupBy(t => t.Name)
-                .Select(g => g.First());
-            var possibleTypes = interfaces
+                .Select(g => g.First());                
+            var myPossibleTypes = interfaces
                 .Where(t => !t.IsIgnored)
                 .SelectMany(t => t.PossibleTypes)
-                .GroupBy(t => t.Name)
-                .Select(g => g.First());
+                .Select(x => OmitNonRootTypes(x))
+                .Where(x => x != null)
+                .ToArray();
             var schema = new Schema(new FuncDependencyResolver(DeriveTypeFromTypeInfo))
             {
                 Query = DeriveOperationType(schemaInfo.Query),
                 Mutation = DeriveOperationType(schemaInfo.Mutation),
                 Subscription = DeriveOperationType(schemaInfo.Subscription),
             };
-            schema.RegisterTypes(possibleTypes
-                .Where(t => !t.IsIgnored && !t.Interfaces.Any(i => i.IsIgnored))
-                .Select(t => DeriveType(t).GetType())
-                .ToArray());
+            // Instead of using Type[] we are now using IGraphType[] for the schema registration
+            schema.RegisterTypes(myPossibleTypes);
```

Comment:
By omitting `!t.Interfaces.Any(i => i.IsIgnored)`, does that mean that you're no longer checking the underlying interfaces, and whether any of those are marked as ignored?